packages/api/src/client/object/fetchPageOrThrow.ts

```diff
@@ -74,14 +75,15 @@ export async function fetchPageOrThrow<
   const r = await loadObjectSetV2(
     createOpenApiRequest(
       client.stack,
-      client.fetchJson as typeof fetch,
+      client.fetch as typeof fetch,
     ),
     client.ontology.metadata.ontologyApiName,
     body,
   );
 
-  return {
-    data: r.data.map((d) => ({ ...d, __name: objectType })),
-    nextPageToken: r.nextPageToken,
-  } as any;
+  convertWireToOsdkObjects(client, objectType, r.data);
+
+  // any is okay here because we have properly converted the wire objects via prototypes
+  // which don't type out correctly.
```

Comment:
The function return type is defined but the definition has a `NOOP<>` in it if you didn't mean to have it there.

Reply:
NOOP can force the mapped types to resolve out. I'm not sure if we want it or not but it's nice to see a small simple object.

Reply:
Maybe just rename it to `Materialize<>` or something then since it isn't a NOOP :)